iam.tf

```diff
@@ -0,0 +1,51 @@
+resource "aws_iam_role" "data_pipeline_iam_role" {
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com",
+        "Service": "s3.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "data_pipeline_archive" {
+  role = "${aws_iam_role.data_pipeline_iam_role.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": ["s3:ListBucket"],
+      "Resource": "${var.archive_bucket}"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:PutObject",
+        "s3:ListObject"
+      ],
+      "Resource": "${var.archive_bucket}/*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": "kms:Decrypt",
+      "Resource": "${var.bucket_key}}"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_instance_profile" "data_pipeline" {
+  role = "${aws_iam_role.data_pipeline_iam_role.arn}"
```

Comment:
I had to change this to `....name` instead of `arn` in the other commit since tf didn't like an arn